[#86] Criado readme em ingles (#87)
salomaoluiz/BuyIt: README_ENGLISH.md

# Project Buy It
[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=salomaoluiz_BuyIt&metric=alert_status)](https://sonarcloud.io/dashboard?id=salomaoluiz_BuyIt)

> This project has be created as a study of case several at development points. If you want use this project to study or commercially, feel free, as long as you follow the [LICENSE](https://github.com/salomaoluiz/BuyIt/blob/main/LICENSE) rules. Most of this project has the descriptions in portuguese, because this way is more easily to a brazilian starter developer understant without language trouble, just the code is fully in english.

Reviewer: ```suggestion
This project has been created as a case study to test several development points. If you want to use this project to study and learn or for commercial purpose, feel free, as long as you follow the [LICENSE](https://github.com/salomaoluiz/BuyIt/blob/main/LICENSE) rules. Most of the project description is in Portuguese, so as to help Brazilian starter developer understand without any language barrier, but the code is written in English and will not create any hindrance in understanding the code itself.
```

## Firebase

> For this project is used the Firebase as a BAS (Backend as a Service), soo you need realize some settings in your project before start.

Reviewer: ```suggestion
This project uses `Firebase` as a BAS (Backend as a Service), so in order to run the project, you need to configure these settings in code.
```

> First, you need create a project in Firebase (Two projects if you wanna separe the development environment of production environment), configure the project to Android and add the **google-services.json** in folders *./android/app/src/development* and *./android/app/src/production*.

Reviewer: ```suggestion
First, you need create a project in `Firebase` (Two projects if you want to separate the development environment from production environment), to configure the project for Android, add the **google-services.json** in folders *./android/app/src/development* and *./android/app/src/production*.
```

> Realized this initial settings, you need adjust the ADMOB for the ads, to configure this is required create 3 files, with each one theys with the following informations:

Reviewer: ```suggestion
After that, you need to adjust the *ADMOB* for the ads, to configure this, you need to create 3 files, each file with the following information:
```

Reviewer: @salomaoluiz There is no information, where these 3 files need to be created, I mean in which folder these files should be?

Author: These files need to be created in root folder, together with *package.json*.

Author: Look if this way is better.

Reviewer: I have updated my edit, mentioning the three files need to be created in root folder

Author: Nice, I will change, thanks.

- **firebase.json**
```
{
  "react-native": {
    "admob_android_app_id": "APP_ID_DO_ADMOB_ANDROID",
    "admob_ios_app_id": "APP_ID_DO_ADMOB_IOS"
  }
}
```
- **.env.dev**
```
DEFAULT_ENVIRONMENT=development
ADMOB_APP_ID=ADMOB_APP_ID
ADMOB_BANNER_ID_DEFAULT=ADMOB_BANNER_ID_DEFAULT
ADMOB_BANNER_ID_CAMPAING=
ADMOB_INTERSTITIAL_ID_DEFAULT=ADMOB_INTERSTITIAL_ID_DEFAULT
ADMOB_INTERSTITIAL_ID_CAMPAING=
```

- **.env.prod**
```
DEFAULT_ENVIRONMENT=production
ADMOB_APP_ID=ADMOB_APP_ID
ADMOB_BANNER_ID_DEFAULT=ADMOB_BANNER_ID_DEFAULT
ADMOB_BANNER_ID_CAMPAING=
ADMOB_INTERSTITIAL_ID_DEFAULT=ADMOB_INTERSTITIAL_ID_DEFAULT
ADMOB_INTERSTITIAL_ID_CAMPAING=
```

> Look, in spite of has a `ADMOB BANNER _ID_*` in the development environment, while you are using the project without release mode, it will show only the test ads.

Reviewer: ```suggestion
> **Note:**, In spite having `ADMOB BANNER _ID_*` in the development environment, while you are using the project in development mode, it will only show test ads.
```

## Creating a build

> To create the project build, has all dependencies installed using the `yarn install` following execute `yarn android:dev` to start in debug mode using the development lane. To generate a app in debug using the production lane is required has a **keystore** configured and in your *$HOME*, after configured it with the name *buyit.keystore* execute in project root `yarn android:prod`.

Reviewer: ```suggestion
To build the project, install all the dependencies using command `yarn install` after that execute `yarn android:dev` to start the app in debug mode using the development lane. To generate the app in debug mode using the production lane requires a **keystore** configured and in your *$HOME*, after configured it with the name *buyit.keystore* execute in project root `yarn android:prod`.
```

> ## After create a build, my app open and close, what I do?
> If you create a build following the passed instructions and when you go open the app in emulator/device it open and close, go to folder *./android* and execute `./gradlew clean` and try realize the build again.

Reviewer: ```suggestion
## After creating a build, app closes after launch, what should I do?
If you created the build following above instructions and when you launch the app in emulator/device it opens and closes immediately, go into the folder *./android* in *terminal* and execute `./gradlew clean` and try to build the app again.
```
